shift_detector/checks/FrequentItemRulesCheck.py

```diff
@@ -40,21 +23,38 @@ class FrequentItemsetCheck(Check):
         in this value other than that it seems sufficiently reasonable.
     """
 
-    def __init__(self, min_support=0.01, min_confidence=0.15):
+    def __init__(self, min_support=0.01, min_confidence=0.15, rule_limit=5):
         self.min_support = min_support
         self.min_confidence = min_confidence
+        self.rule_limit = rule_limit
 
-    def run(self, store: Store) -> FrequentItemsetReport:
+    def run(self, store: Store) -> Report:
         """
         Calculate frequent rules, compress them and create a
         FrequentItemsetReport.
         :param store: the Store
         :return: FrequentItemsetReport
         """
-        df1 = store.df1
-        df2 = store.df2
+        precalculation_result = store[FrequentItemsetPrecalculation()]
+        compressed_rules = precalculation_result['compressed_rules']
+        examined_columns = precalculation_result['examined_columns']
```

Comment:
Why are you retrieving the examined columns from the precalculation? You should be able to retrieve it within the check and simplify the return value of the precalc from a dict to the compressed rules only